app/services/update_user.rb

```diff
@@ -46,12 +46,17 @@ def phone_attributes
   end
 
   def made_default_at_date
-    return current_made_default_at if attributes[:otp_make_default_number].blank?
-    Time.zone.now
+    otp_make_default_number = attributes[:otp_make_default_number]
+
+    if otp_make_default_number.blank? || otp_make_default_number == 'false'
+      current_made_default_at
+    else
+      Time.zone.now
+    end
```

Comment:
what about a positive check?
```suggestion
    if attributes[:otp_make_default_number].to_s == 'true'
      Time.zone.now
    else
      current_made_default_at
    end
```

Reply:
I like that better